contracts/governance/Vesting/VestingCreator.sol

```diff
@@ -0,0 +1,288 @@
+pragma solidity ^0.5.17;
+
+import "../../interfaces/IERC20.sol";
+import "../../utils/AdminRole.sol";
+import "./VestingRegistryLogic.sol";
+import "./VestingLogic.sol";
+import "../../openzeppelin/SafeMath.sol";
+
+contract VestingCreator is AdminRole {
+	using SafeMath for uint256;
+
+	///@notice Boolean to check both vesting creation and staking is completed for a record
+	bool vestingCreated;
+
+	/// @notice 2 weeks in seconds.
+	uint256 constant TWO_WEEKS = 1209600;
+
+	///@notice the SOV token contract
+	IERC20 public SOV;
+
+	///@notice the vesting registry contract
+	VestingRegistryLogic public vestingRegistryLogic;
+
+	///@notice Holds Vesting Data
+	struct VestingData {
+		address tokenOwner;
+		uint256 amount;
+		uint256 cliff;
+		uint256 duration;
+		bool governanceControl; ///@dev true - tokens can be withdrawn by governance
+		uint256 vestingCreationType;
+	}
+
+	///@notice list of vesting to be processed
+	VestingData[] public vestingDataList;
+
+	event SOVTransferred(address indexed receiver, uint256 amount);
+	event TokensStaked(address indexed vesting, address indexed tokenOwner, uint256 amount);
+	event VestingDataRemoved(address indexed caller, address tokenOwner);
+	event DataCleared(address indexed caller);
+
+	constructor(address _SOV, address _vestingRegistryLogic) public {
+		require(_SOV != address(0), "SOV address invalid");
+		require(_vestingRegistryLogic != address(0), "Vesting registry address invalid");
+
+		SOV = IERC20(_SOV);
+		vestingRegistryLogic = VestingRegistryLogic(_vestingRegistryLogic);
+	}
+
+	/**
+	 * @notice transfers SOV tokens to given address
+	 * @param _receiver the address of the SOV receiver
+	 * @param _amount the amount to be transferred
+	 */
+	function transferSOV(address _receiver, uint256 _amount) public onlyOwner {
+		require(_receiver != address(0), "receiver address invalid");
+		require(_amount != 0, "amount invalid");
+
+		require(SOV.transfer(_receiver, _amount), "transfer failed");
+		emit SOVTransferred(_receiver, _amount);
+	}
+
+	/**
+	 * @notice adds vestings to be processed to the list
+	 * @dev if account doesn't have another vesting of the same type vesting data will be added to vestingDataList
+	 */
+	function addVestings(
+		address[] memory _tokenOwners,
+		uint256[] memory _amounts,
+		uint256[] memory _cliffs,
+		uint256[] memory _durations,
+		bool[] memory _governanceControls,
+		uint256[] memory _vestingCreationTypes
```

Comment:
The data will be available from a CSV file and hence it was planned this way. The front end is implemented and tested and hence it will be difficult to rework.